_posts/2016-12-06-future.md

```diff
@@ -0,0 +1,444 @@
+
+There is a lot of confusion around how plugins/presets interact regarding ordering. This results in bugs and issues with the config which require users to place plugins before/after others in a non-intuitive way.
+
+This is a fundamental change to the core of Babel which will take some work to figure out.
+
+### Versioning
+
+Since Babel 6 we've used a ["fixed" mode of versioning](https://github.com/lerna/lerna#how-it-works) via Lerna. This is what allows us to release multiple packages at the same time all under the same version (if that package changes). This is nice because you don't have to manually set a version for each package but everything moves together. The only time this is a pain is when one on the packages makes a breaking change: then every package also will bump major versions.
+
+This is explained more in [babel/notes](https://github.com/babel/notes/blob/master/2016-07/july-31.md#future-of-babels-release-process-and-its-ecosystem) but we still need to figure out the best plan of action for the project.
+
+What happens when we need to update a Stage 0 spec to Stage 1 and it's a breaking change to the parser? Are we just going to bump the major version, wait a batch some changes up, or figure out how to do it via multiple versions of plugins?
+
+[Discussion Issue](https://github.com/babel/babel/issues/4950)
+
+#### Changing the Mindset Around Stage X Presets
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">My rule of thumb on how I decide what future features to transpile:<br>&quot;Could I reasonably codemod this if it changes?&quot;<br>Don&#39;t do it otherwise.</p>&mdash; Kent C. Dodds (@kentcdodds) <a href="https://twitter.com/kentcdodds/status/803815749416456196">November 30, 2016</a></blockquote>
+<script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+Relevant Issues:
+
+- [Should we rename the stage x presets to explain more that they are "experimental"](https://github.com/babel/babel/issues/4914)
+
+#### Speed
+
+Performance is a feature! Other things can be more of a priority at times (bug fixes, spec compliancy, etc) but it's still important in a few different aspects.
+
+- How can we reduce the size/time of install, especially for a project with multiple packages? (this is helped by [yarn](yarnpkg.com))
+- How can we parse faster?
+- How can we make faster plugins (and measure them individually)?
+- How can we generate the transformed code faster?
+- How can we generate code that runs fast in the browser (https://fhinkel.github.io/six-speed/)?
+
+If you are one to read the compiled output and see issues, then report it and ask for help in making a PR!
+
+Previous Issues:
+
+- [Speeeeed](https://github.com/babel/babel/issues/1486)
+- [Amazing PR to improve babel-generator by @gzzhanghao](https://github.com/babel/babel/pull/3283)
+
+### TypeScript Support
```

Comment:
Maybe worth making it clear these are just ideas? This makes it sound like something we're planning specifically.